pallets/xcmp-handler/src/lib.rs

```diff
@@ -228,6 +228,32 @@ pub mod pallet {
 			Ok(instructions)
 		}
 
+		fn get_local_xcm(
+			asset: MultiAsset,
+			destination: MultiLocation,
+		) -> Result<xcm::latest::Xcm<<T as pallet::Config>::RuntimeCall>, DispatchError> {
+			let reserve =
+				T::ReserveProvider::reserve(&asset).ok_or(Error::<T>::InvalidAssetLocation)?;
+			let local_xcm = if reserve == MultiLocation::here() {
+				Xcm(vec![
+					WithdrawAsset::<<T as pallet::Config>::RuntimeCall>(asset.into()),
+					DepositAsset::<<T as pallet::Config>::RuntimeCall> {
+						assets: Wild(All),
+						beneficiary: destination,
+					},
+				])
+			} else if reserve == destination {
+				Xcm(vec![
+					WithdrawAsset::<<T as pallet::Config>::RuntimeCall>(asset.clone().into()),
+					BurnAsset::<<T as pallet::Config>::RuntimeCall>(asset.into()),
+				])
+			} else {
+				return Err(Error::<T>::UnsupportedFeePayment.into())
+			};
+
+			Ok(local_xcm)
+		}
```

Comment:
okay, let’s make the logic simpler with some restructuring. Because the local_xcm and target_xcm always go in pair, let’s not write `if reserve == MultiLocation::here()` twice separately. Instead,

```
let reserve = T::ReserveProvider::reserve(&asset).ok_or(Error::<T>::InvalidAssetLocation)?;
if reserve == MultiLocation::here() {
let local_xcm = ...
let target_xcm = ...
Ok((local_xcm, target_xcm))
} else if reserve == destination {
let local_xcm = ...
let target_xcm = ...
Ok((local_xcm, target_xcm))
} else {}

```